### PR TITLE
OCPBUGS-5790: Allow cluster to be upgraded even if using multiple datacenters

### PR DESCRIPTION
**Pull request by gnufied**
Fixes https://issues.redhat.com/browse/OCPBUGS-5790

This is basically a manual cherry-pick of https://github.com/openshift/vmware-vsphere-csi-driver-operator/pull/123 because automatic cherry-pick was throwing tons of conflicts.

### CODE REVIEW

File: pkg/operator/utils/topology.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/legacy-cloud-providers/vsphere"
+)
+
+func GetDatacenters(config *vsphere.VSphereConfig) ([]string, error) {
```
Comment (dobsonj) [inline pkg/operator/utils/topology.go:11]: Don't you need the changes in pkg/operator/vspherecontroller/checks/check_nodes.go too?
https://github.com/openshift/vmware-vsphere-csi-driver-operator/pull/123/files#diff-872043e7b350e043c36bbae56b122ac24f16df367350b832376f0e3a2a23582eL239-R240

Comment (gnufied) [reply pkg/operator/utils/topology.go:11]: Good catch! Will fix.

Comment (gnufied) [reply pkg/operator/utils/topology.go:11]: fixed.

**Pull request merged**